### PR TITLE
Nixos service fixes

### PR DESCRIPTION
**Pull request by jbgi**

### CODE REVIEW

File: nix/nixos/cardano-node.nix [hunk 1]
```diff
@@ -36,12 +37,32 @@ in {
...
+      };
+
+      signing-key = mkOption {
+        type = types.path;
+        description = ''
+          Signing key
+        '';
+      };
+
+      delegation-certificate = mkOption {
```
Comment (jbgi) [inline nix/nixos/cardano-node.nix]: should the delegation certificate be mandatory?

File: nix/nixos/cardano-node.nix [hunk 2]
```diff
@@ -118,26 +138,25 @@ in {
...
-            port = builtins.head (builtins.tail (builtins.tail s));
-          }) cfg.producers-peers;
-        }]);
-
-      in ''
-        START_TIME=${if (cfg.system-start-time != null) then cfg.system-start-time else "`date \"+%Y-%m-%d 00:00:00\"`"}
-        ${cfg.package}/bin/cardano-node --log-config ${cfg.logger.config-file} --system-start "$START_TIME" --slot-duration ${builtins.toString cfg.slot-duration} node --topology ${topology} --${cfg.consensus-protocol} --node-id ${builtins.toString cfg.node-id} --host ${cfg.host} --port ${builtins.toString cfg.port}
-      '';
+      script = ''
+        GENESIS_START_TIME=$(${pkgs.jq}/bin/jq '.startTime' < ${cfg.genesis-file})
```
Comment (jbgi) [inline nix/nixos/cardano-node.nix]: Is it ok to get the start time defined in genesis file and use it as argument to `--system-start`?

File: nix/nixos/cardano-node.nix (hunk 1 above)
Comment (dcoutts) [reply nix/nixos/cardano-node.nix]: It's only required for core nodes that sign blocks. We want to run relays too.

File: nix/nixos/cardano-node.nix (hunk 2 above)
Comment (dcoutts) [reply nix/nixos/cardano-node.nix]: Good question. You'll need to pester us for a proper answer on that. I think for mainnet the answer is yes, but for dev and testing I'm not so sure. Ask @mrBliss

**Comment by dcoutts** [comment]
@jbgi should these patches be squashed before we merge, or can one invoke bors to do that?

**Pull request merged**